src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -88,103 +138,1165 @@ static DamageTable D_80A31B1C = {
+                    EnSlime_ReactToDamage(this, play, true);
+                }
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F9A0.s")
+void EnSlime_Update(Actor* thisx, PlayState* play) {
+    EnSlime* this = THIS;
+    s32 pad;
+    Player* player = GET_PLAYER(play);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2FA88.s")
+    EnSlime_CheckACHit(this, play);
+    this->actionFunc(this, play);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2FB60.s")
+    // Note: Does not match if you use this->actor instead of thisx
+    thisx->shape.shadowAlpha = this->iceBlockTimer;
+    if (this->iceBlockTimer == 0xFF) {
+        if (thisx->scale.y > 0.0001f) {
+            thisx->targetArrowOffset = 60.0f / thisx->scale.y;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2FBA0.s")
+        if (this->collider.base.ocFlags1 & OC1_ON) {
+            Actor_MoveWithGravity(thisx);
+            Actor_UpdateBgCheckInfo(play, thisx, 20.0f, 35.0f, 40.0f, 0x1D);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2FD94.s")
+        Collider_UpdateCylinder(thisx, &this->collider);
+        this->collider.dim.radius = sCylinderInit.dim.radius * (100.0f * thisx->scale.z);
+        this->collider.dim.height = sCylinderInit.dim.height * (100.0f * thisx->scale.y);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2FE38.s")
+        if (this->actionFunc == EnSlime_UpdateJump) {
+            CollisionCheck_SetAT(play, &play->colChkCtx, &this->collider.base);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A30018.s")
+        if (this->collider.base.acFlags & AC_ON) {
+            CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A30344.s")
+        if (player->stateFlags3 & PLAYER_STATE3_100) {
+            this->collider.base.ocFlags1 &= ~OC1_TYPE_PLAYER;
+        } else {
+            this->collider.base.ocFlags1 |= OC1_TYPE_PLAYER;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A30454.s")
+        if (this->collider.base.ocFlags1 & OC1_ON) {
+            CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A304B8.s")
+        Actor_SetFocus(thisx, 15.0f);
+        if (this->effectAlpha > 0.0f) {
+            if (this->damageEffectType != ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX) {
+                Math_StepToF(&this->effectAlpha, 0.0f, 0.05f);
+                this->effectScale = (this->effectAlpha + 1.0f) * 0.2f;
+                this->effectScale = CLAMP_MAX(this->effectScale, 0.4f);
+            } else if (Math_StepToF(&this->frozenSteamScale, 0.4f, 0.01f) == 0) {
+                func_800B9010(thisx, 0x20B2);
```

Comment:
```suggestion
                func_800B9010(&this->actor, NA_SE_EV_ICE_FREEZE - SFX_FLAG);
```
This is actually a sfxId. Also, we prefer `&this->actor` unless `thisx` is needed somewhere in function to match